spec/content/automate/ManageIQ/Service/Generic/StateMachines/GenericLifecycle.class/__methods__/check_completed_spec.rb

```diff
@@ -71,4 +65,105 @@
     end
     it_behaves_like "check_completed_error"
   end
+
+  context "retry ttl" do
+    let(:service_ansible_tower) { FactoryGirl.create(:service_ansible_tower, :job_template => job_template, :options => config_info_options) }
+    let(:config_info_options) do
+      {
+        :config_info => {
+          :provision  => {
+            :execution_ttl => ttl
+          },
+          :retirement => {
+            :execution_ttl => ttl
+          }
+        }
+      }
+    end
+    shared_examples_for "#ttl" do
+      it "check" do
+        allow(svc_service).to receive(:check_completed).and_return([false, "retry"])
+
+        described_class.new(ae_service).main
+        expect(ae_service.root['ae_retry_interval']).to eq(ae_retry_interval)
+      end
+    end
+
+    context "provision tests " do
+      let(:root_object) do
+        Spec::Support::MiqAeMockObject.new('service' => svc_service, 'service_action' => 'Provision', 'ae_state_max_retries' => 100)
+      end
+
+      context "300 ttl, 100 retries eq interval 3" do
+        let(:ae_retry_interval) { 3.minutes }
+        let(:ttl) { 300 }
+        it_behaves_like "#ttl"
+      end
+
+      context "600 ttl, 100 retries eq interval 6" do
```

Comment:
@tinaafitz whats the difference between these different scenarios could we just use 1 test. maybe the 300 tel or the 600 tel